File: src/main/resources/output/IFC4X3_csharp/Entities/IfcMaterialDefinition/IfcMaterialProfile/IfcMaterialProfile.cs
public class IfcMaterialProfile : IfcMaterialDefinition
{
    public IfcLabel Name { get; set; }
    public IfcText Description { get; set; }
    public IfcMaterial Material { get; set; }
    public IfcProfileDef Profile { get; set; }
    public IfcInteger Priority { get; set; }
    public IfcLabel Category { get; set; }

    // === EXTENDED BY ===
    // IfcMaterialProfileWithOffsets

    // === INVERSE CLAUSES ===
    // ToMaterialProfileSet : IfcMaterialProfileSet FOR MaterialProfiles

    // === WHERE CLAUSES ===
    // NormalizedPriority : NOT(EXISTS(Priority)) OR {0 <= Priority <= 100}
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcMaterialDefinition/IfcMaterialProfile/IfcMaterialProfile.vb
Public Class IfcMaterialProfile Inherits IfcMaterialDefinition
    Public Name As IfcLabel
    Public Description As IfcText
    Public Material As IfcMaterial
    Public Profile As IfcProfileDef
    Public Priority As IfcInteger
    Public Category As IfcLabel

    ' === EXTENDED BY ===
    ' IfcMaterialProfileWithOffsets

    ' === INVERSE CLAUSES ===
    ' ToMaterialProfileSet : IfcMaterialProfileSet FOR MaterialProfiles

    ' === WHERE CLAUSES ===
    ' NormalizedPriority : NOT(EXISTS(Priority)) OR {0 <= Priority <= 100}
End Class
